api/handlers_spaces.go

```diff
@@ -37,15 +85,45 @@ func (s *server) SpaceGetHandler(w http.ResponseWriter, r *http.Request) {
 	}
 	log.Debugf("found cost explorer result cache %+v", *resultCache)
 
-	spaceID := vars["space"]
-	log.Debugf("getting costs for space %s", spaceID)
+	// vars for checking input times parse and are valid
+	var timeValidity bool
+	var start string
+	var end string
+	var err error
 
-	y, m, d := time.Now().Date()
+	// Did we get cost-explorer start and end times on the API?
+	// set defaults, else verify times given on API
+	if endTime == "" || startTime == "" {
+		start, end, timeValidity = getTimeDefault()
+	} else {
+		start, end, timeValidity, err = getTimeAPI(startTime, endTime)
+		if err != nil {
+			// unwrap errors
+			if aerr, ok := errors.Cause(err).(apierror.Error); ok {
+				switch aerr.Code {
+				case apierror.ErrForbidden:
+					w.WriteHeader(http.StatusForbidden)
+				case apierror.ErrNotFound:
+					w.WriteHeader(http.StatusNotFound)
+				case apierror.ErrConflict:
+					w.WriteHeader(http.StatusConflict)
+				case apierror.ErrBadRequest:
+					w.WriteHeader(http.StatusBadRequest)
+				case apierror.ErrLimitExceeded:
+					w.WriteHeader(http.StatusTooManyRequests)
+				default:
+					w.WriteHeader(http.StatusInternalServerError)
+				}
+				w.Write([]byte(aerr.Message))
+			} else {
+				w.WriteHeader(http.StatusInternalServerError)
+				w.Write([]byte(err.Error()))
```

Comment:
`handleError()` does all of this already, I think you just need to call it here

https://github.com/YaleSpinup/cost-api/blob/master/api/handlers.go#L48